api/v1alpha1/dnshealthcheckprobe_types.go

```diff
@@ -97,14 +97,25 @@ type DNSHealthCheckProbeList struct {
 	Items           []DNSHealthCheckProbe `json:"items"`
 }
 
+func (p *DNSHealthCheckProbe) GetIPAddresses() []string {
+	if net.ParseIP(p.Spec.Address) != nil {
+		return []string{
+			p.Spec.Address,
+		}
+	}
+	ips, _ := net.LookupHost(p.Spec.Address)
```

Comment:
why do we ignore an error here?

Reply:
This whole function seems to not be needed. It can be replaced just be `net.LookupHost()` call (it checks for address to be IP inside). Not sure why it was decided to ignore error. I see not harm in checking for it.